pkg/config/creds.go

```diff
@@ -45,10 +60,12 @@ func getCredentials(decryptor secrets.Decryptor) (Creds, error) {
 
 	if len(encrypted) == 0 {
 		// Nothing to decrypt
-		return Creds{APIKey: apiKey, AppKey: appKey}, nil
+		creds := Creds{APIKey: apiKey, AppKey: appKey}
+		cm.cacheCreds(creds)
+		return creds, nil
 	}
 
-	decrypted, err := decryptor.Decrypt(encrypted)
+	decrypted, err := cm.secretBackend.Decrypt(encrypted)
 	if err != nil {
 		return Creds{}, err
 	}
```

Comment:
The code seems structured to support the following case:
* `decrypted, err := cm.secretBackend.Decrypt(encrypted)` returns `err == nil` but
* `apiKey` is in `decrypted`, but not `appKey`.

Can this happen in practice ?
That would mean that one secret could be found, but not the other one.

In such a case, we would cache a decrypted api key but the app key would be cached encrypted and we would never recover automatically.

Reply:
`cm.secretBackend.Decrypt(encrypted)` returns an error if it cannot decrypt all the elements in `encrypted`.
Does this answer to your question?

Reply:
Yes.
If `cm.secretBackend.Decrypt(encrypted)` cannot return some secrets but not all, it’s OK.

Reply:
it's already the case 👍